cli/azd/pkg/auth/cache.go

```diff
@@ -36,6 +48,35 @@ func (a *msalCacheAdapter) Replace(ctx context.Context, cache cache.Unmarshaler,
 		return err
 	}
 
+	// In msal v1.0, keys were stored with mixed casing; in v1.1., it was changed to lower case.
+	// This handles upgrades where we have a v1.0 cache, and we need to convert it to v1.1,
+	// by normalizing the appropriate key entries.
+	c := map[string]json.RawMessage{}
+	if err = json.Unmarshal(val, &c); err == nil {
+		for _, contractKey := range contractFields {
+			if _, ok := c[contractKey]; ok {
+				msg := []byte(c[contractKey])
+				inner := map[string]json.RawMessage{}
+
+				if err := json.Unmarshal(msg, &inner); err == nil {
```

Comment:
do we want to ignore an error here?

Reply:
Yes, but I've added additional logging. We're dealing with a cache format that may change over time. Thus, it's better to not error out if something unexpected happens.